techlibs/gatemate/synth_gatemate.cc

```diff
@@ -0,0 +1,393 @@
+		}
+
+		if (check_label("map_bram", "(skip if '-nobram')"))
+		{
+			if (!nobram) {
+				run("memory_bram -rules +/gatemate/brams.txt");
+				run("setundef -zero -params "
+					"t:$__CC_BRAM_CASCADE "
+					"t:$__CC_BRAM_40K_SDP t:$__CC_BRAM_20K_SDP "
+					"t:$__CC_BRAM_20K_TDP t:$__CC_BRAM_40K_TDP "
+				);
+				run("techmap -map +/gatemate/brams_map.v");
+				run("gatemate_bramopt");
+			}
+		}
+
+		if (check_label("map_ffram"))
+		{
+			run("opt -fast -mux_undef -undriven -fine");
+			run("memory_map");
+			run("opt -undriven -fine");
+		}
+
+		if (check_label("map_gates"))
+		{
+			std::string techmap_args = "";
+			if (!noaddf) {
+				techmap_args += " -map +/gatemate/arith_map.v";
+			}
+			run("techmap -map +/techmap.v " + techmap_args);
+			run("opt -fast");
+			if (retime) {
+				run("abc -dff -D 1", "(only if -retime)");
+			}
+		}
+
+		if (check_label("map_io", "(skip if '-noiopad')"))
+		{
+			if (!noiopad) {
+				run("iopadmap -bits "
+					"-inpad CC_IBUF Y:I "
+					"-outpad CC_OBUF A:O "
+					"-toutpad $__toutpad OE:A:O "
+					"-tinoutpad $__tinoutpad OE:Y:A:IO"
+				);
+				run("techmap -map +/gatemate/iob_map.v");
+				run("clean");
+			}
+		}
+
+		if (check_label("map_regs"))
+		{
+			run("opt_clean");
+			run("dfflegalize -cell $_DFFE_????_ x -cell $_DLATCH_???_ x");
+			run("techmap -map +/gatemate/reg_map.v");
+			run("opt_expr -mux_undef");
+			run("simplemap");
+			run("opt_clean");
+		}
+
+		if (check_label("map_muxs"))
+		{
+			run("abc -g simple"); // optimize design before muxcover
+			std::string muxcover_args;
+			if (!nomx4) {
+				muxcover_args += stringf(" -mux4");
+			}
+			if (!nomx8) {
+				muxcover_args += stringf(" -mux8");
+			}
+			run("muxcover " + muxcover_args);
+			run("opt -full");
+			run("techmap -map +/gatemate/mux_map.v");
+		}
+
+		if (check_label("map_luts"))
+		{
+			std::string abc_args = " -dress -lut 4";
+			if (dff) {
+				abc_args += " -dff";
+			}
+			run("abc " + abc_args);
+			run("clean");
+		}
+
+		if (check_label("map_cells"))
+		{
+			run("techmap -map +/gatemate/lut_map.v");
+			run("clean");
+		}
+
+		if (check_label("map_bufg", "(skip if '-noclkbuf')"))
+		{
+			if (!noclkbuf) {
+				run("clkbufmap -buf CC_BUFG O:I");
+				run("clean");
+			}
+		}
+
+		if (check_label("check"))
+		{
+			run("hierarchy -check");
+			run("stat -width");
+			run("check -noinit");
+			run("blackbox =A:whitebox");
+		}
+
+		if (check_label("vlog"))
+		{
+			run("opt_clean -purge");
+			if (!vlog_file.empty() || help_mode) {
+				run(stringf("write_verilog -noattr %s", help_mode ? "<file-name>" : vlog_file.c_str()));
+			}
+		}
+
+		if (check_label("edif"))
+		{
+			if (!edif_file.empty() || help_mode) {
+				run(stringf("write_edif %s", help_mode ? "<file-name>" : edif_file.c_str()));
+			}
+		}
+
+		if (check_label("json"))
+		{
+			if (!json_file.empty() || help_mode) {
+				run(stringf("write_json %s", help_mode ? "<file-name>" : json_file.c_str()));
+			}
+		}
+
+		if (check_label("blif"))

```

Comment:
Same as for EDIF: do you need this? Sure, VPR uses BLIF input, but do you have a VPR toolchain?